pkg/test/expectations/expectations.go

```diff
@@ -568,3 +568,14 @@ func ExpectMakeNodesAndNodeClaimsInitializedAndStateUpdated(ctx context.Context,
 		ExpectReconcileSucceeded(ctx, nodeClaimStateController, client.ObjectKeyFromObject(m))
 	}
 }
+
+func ExpectEvicted(ctx context.Context, c client.Client, pods ...*v1.Pod) {
```

Comment:
I think this should just have the evict() call in here, and call an ExpectDeleted() or ExpectDeleting()